uaclient/util.py

```diff
@@ -615,3 +615,52 @@ def is_config_value_true(config: "Dict[str, Any]", path_to_value: str):
                 value=value_str,
             )
         )
+
+
+def depth_first_merge_overlay_dict(base_dict, overlay_dict):
+    """Merge the contents of overlay dict into base_dict not only on top-level
+    keys, but on all on the depths of the overlay_dict object. For example,
+    using these values as entries for the function:
+
+    base_dict = {"a": 1, "b": {"c": 2, "d": 3}}
+    overlay_dict = {"b": {"c": 10}}
+
+    Should update base_dict into:
+
+    {"a": 1, "b": {"c": 10, "d": 3}}
+
+    @param base_dict: The dict to be updated
+    @param overlay_dict: The dict with information to be added into base_dict
+    """
+
+    def get_id_key(key):
+        id_map = {"availableResources": "name", "resourceEntitlements": "type"}
+
+        return id_map.get(key)
```

Comment:
This doesn't need to be a method. Let's just define a global dict

```suggestion
MERGE_ID_KEY_MAP = {"availableResources": "name", "resourceEntitlements": "type"}
```